src/ga_session.cpp

```diff
@@ -3702,6 +3702,76 @@ namespace sdk {
         return sign_ga_transaction(*this, details);
     }
 
+    nlohmann::json ga_session::sign_psbt(const nlohmann::json& details)
+    {
+        nlohmann::json result = details;
+        const std::string txhex = extract_tx_from_psbt(details.at("psbt"));
+        const nlohmann::json tx_details = { { "transaction", txhex } };
+        const auto flags = tx_flags(m_net_params.is_liquid());
+        wally_tx_ptr tx = tx_from_hex(txhex, flags);
+
+        // Clear utxos and fill it with the one that will be signed
+        std::vector<nlohmann::json> inputs;
+        inputs.reserve(tx->num_inputs);
+        for (size_t i = 0; i < tx->num_inputs; ++i) {
+            bool found = false;
```

Comment:
Instead of using `found`, have a variable `input_utxo = nlohmann::json::object()`. when the utxo is found in `result`, `std::move` the utxo into `input_utxo`, and at the end of the loop, unconditionally add `input_utxo` to `inputs` (you will need to check `!uxto.empty()` before matching the txhash/vout). This makes the logic simpler and avoids a bunch of copies

Reply:
Changed as suggested, thanks